runtime/sema/type.go

```diff
@@ -2497,6 +2504,32 @@ func getArrayMembers(arrayType ArrayType) map[string]MemberResolver {
 				)
 			},
 		}
+
+		members[ArrayTypeToConstantSizedFunctionName] = MemberResolver{
+			Kind: common.DeclarationKindFunction,
+			Resolve: func(memoryGauge common.MemoryGauge, identifier string, targetRange ast.Range, report func(error)) *Member {
+				elementType := arrayType.ElementType(false)
+
+				if elementType.IsResourceType() {
+					report(
+						&InvalidResourceArrayMemberError{
+							Name:            identifier,
+							DeclarationKind: common.DeclarationKindFunction,
+							Range:           targetRange,
+						},
+					)
+				}
+
+				return NewFunctionMember(
+					memoryGauge,
+					arrayType,
+					insertableEntitledAccess,
+					identifier,
+					ArrayToConstantSizedFunctionType(elementType),
+					arrayTypeToConstantSizedFunctionDocString,
+				)
```

Comment:
This should be a public function, since it doesn't mutate the existing one.
```suggestion
				return NewPublicFunctionMember(
					memoryGauge,
					arrayType,
					identifier,
					ArrayToConstantSizedFunctionType(elementType),
					arrayTypeToConstantSizedFunctionDocString,
				)
```

Reply:
Good catch, didn't notice this during review.

Opened #3031 to fix `toVariableSized`

Reply:
This should also be fixed for `toVariableSized`

Reply:
Thanks!